Content.Shared/SegmentedEntity/SegmentedEntitySystem.cs

```diff
@@ -0,0 +1,306 @@
+                    continue;
+
+                EnsureComp<PhysicsComponent>(segmentUid);
+                EnsureComp<PhysicsComponent>(attachedUid); // Hello I hate tests
+
+                // This is currently HERE and not somewhere more sane like OnInit because HumanoidAppearanceComponent is for whatever
+                // ungodly reason not initialized when ComponentStartup is called. Kill me.
+                var humanoidFactor = TryComp<HumanoidAppearanceComponent>(segment.segment.Lamia, out var humanoid) ? (humanoid.Height + humanoid.Width) / 2 : 1;
+
+                var ev = new SegmentSpawnedEvent(segment.lamia);
+                RaiseLocalEvent(segmentUid, ev, false);
+
+                if (segment.segment.SegmentNumber == 1)
+                {
+                    _transform.SetCoordinates(segmentUid, Transform(attachedUid).Coordinates);
+                    var revoluteJoint = _jointSystem.CreateWeldJoint(attachedUid, segmentUid, id: "Segment" + segment.segment.SegmentNumber + segment.segment.Lamia);
+                    revoluteJoint.CollideConnected = false;
+                }
+                if (segment.segment.SegmentNumber <= segment.segment.MaxSegments)
+                    _transform.SetCoordinates(segmentUid, Transform(attachedUid).Coordinates.Offset(new Vector2(0, segment.segment.OffsetSwitching * humanoidFactor)));
+                else
+                    _transform.SetCoordinates(segmentUid, Transform(attachedUid).Coordinates.Offset(new Vector2(0, segment.segment.OffsetSwitching * humanoidFactor)));
+
+                var joint = _jointSystem.CreateDistanceJoint(attachedUid, segmentUid, id: ("Segment" + segment.segment.SegmentNumber + segment.segment.Lamia));
+                joint.CollideConnected = false;
+                joint.Stiffness = 0.2f;
+            }
+            _segments.Clear();
+        }
+        private void OnInit(EntityUid uid, SegmentedEntityComponent component, ComponentInit args)
+        {
+            EnsureComp<PortalExemptComponent>(uid); //Temporary, remove when Portal handling is added
+            Math.Clamp(component.NumberOfSegments, 2, 18);
+            Math.Clamp(component.TaperOffset, 1, component.NumberOfSegments - 1);
+            SpawnSegments(uid, component);
+        }
+
+        private void OnShutdown(EntityUid uid, SegmentedEntityComponent component, ComponentShutdown args)
+        {
+            if (_net.IsClient)
+                return;
+
+            foreach (var segment in component.Segments)
+            {
+                QueueDel(GetEntity(segment));
+            }
+
+            component.Segments.Clear();
+        }
+
+        /// <summary>
+        ///     TODO: Full Self-Test function that intelligently checks the status of where everything is, and calls whatever
+        ///     functions are appropriate
+        /// </summary>
+        /// <param name="uid"></param>
+        /// <param name="component"></param>
```

Comment:
either write commentaries explaining what this params are or remove this from xaml docs entirely.